pandas/io/parsers/arrow_parser_wrapper.py

```diff
@@ -158,6 +160,19 @@ def read(self) -> DataFrame:
             parse_options=pyarrow_csv.ParseOptions(**self.parse_options),
             convert_options=pyarrow_csv.ConvertOptions(**self.convert_options),
         )
+
+        # Convert all pa.null() cols -> float64
+        # TODO: There has to be a better way... right?
+        cols_to_convert = []
+        new_schema = table.schema
+        for i, type in enumerate(table.schema.types):
+            if pa.types.is_null(type):
+                cols_to_convert.append(i)
+        for i in cols_to_convert:
+            new_schema = new_schema.set(i, new_schema.field(i).with_type(pa.float64()))
```

Comment:
Cant this not be done in the `if pa.types.is_null` block?

Reply:
Thanks.